SharedUtils/src/main/java/com/braintreepayments/api/SynchronousHttpClient.java

```diff
@@ -75,9 +77,13 @@ String request(HttpRequest httpRequest) throws Exception {
 
         try {
             int responseCode = connection.getResponseCode();
-            return parser.parse(responseCode, connection);
+            long endTime = System.currentTimeMillis();
+
+            String responseBody = parser.parse(responseCode, connection);
+
+            return new HttpResponse(startTime, endTime, responseBody);
```

Comment:
Random thought: We should circle back and preserve the response headers here as well. I could see us surfacing PayPal debug IDs in error messages in the near future.